pkg/controller/master-controller-manager/rbac/mapper.go

```diff
@@ -665,21 +665,79 @@ func generateVerbsForNamespacedResource(groupName, resourceKind, namespace strin
 func generateVerbsForNamedResourceInNamespace(groupName, resourceKind, namespace string) ([]string, error) {
 	// special case - only the owners of a project can manipulate secrets in "ssaSecretsNamespaceName" namespace
 	//
-	if namespace == saSecretsNamespaceName || strings.HasPrefix(namespace, resources.KubeOneNamespacePrefix) {
-		switch {
-		case strings.HasPrefix(groupName, OwnerGroupNamePrefix) && resourceKind == secretV1Kind:
-			return []string{"get", "update", "delete"}, nil
-		case strings.HasPrefix(groupName, ProjectManagerGroupNamePrefix) && resourceKind == secretV1Kind:
-			return []string{"get", "update", "delete"}, nil
-		case resourceKind == secretV1Kind:
-			return nil, nil
-		}
+	if !isAcceptedNamespace(namespace) {
+		return nil, fmt.Errorf("unable to generate verbs, unsupported namespace %q given", namespace)
+	}
+	switch resourceKind {
+	case secretV1Kind:
+		return generateVerbsForNamedSecretKindInNamespace(groupName)
+
+	case kubermaticv1.ClusterBackupStorageLocationKind:
+		return generateVerbsForNamedCBSLKindInNamespace(groupName)
 	}
 
 	// unknown group passed
 	return nil, fmt.Errorf("unable to generate verbs for group %q, kind %q and namespace %q", groupName, resourceKind, namespace)
 }
 
+func isAcceptedNamespace(namespace string) bool {
+	if namespace != resources.KubermaticNamespace && !strings.HasPrefix(namespace, resources.KubeOneNamespacePrefix) {
+		return false
+	}
+	return true
+}
```

Comment:
`return namespace == resources.KubermaticNamespace || strings.HasPrefix(namespace, resources.KubeOneNamespacePrefix)`, no need for an if stament. Especially not an if statement with inverted conditions that than, if it works, returns false. 😁